### PR TITLE
Add Item SubType to data structures and DB

### PR DESCRIPTION
**Pull request by mackal**
This can be used by other devs to implement other features now (fishing pole changes, Ff_ItemClass, etc)

### CODE REVIEW

**Comment by mackal** [comment]
@noudess

**Comment by noudess** [comment]
Thank you!  Once merged, I'll remove the sql stuff from my PR and make the remaining changes.

File: common/item_fieldlist.h [hunk 1]
```diff
@@ -203,3 +203,4 @@ F(procname)
 F(wornname)
 F(focusname)
 F(scrollname)
+F(subtype)
```
Comment (mackal) [inline common/item_fieldlist.h:206]: should we order this better? Does it matter?

Comment (noudess) [reply common/item_fieldlist.h:206]: I defer to you.  I'll add the accessor and change my PR tomorrow.

**Pull request merged**